### PR TITLE
Give works to ISBNs through OCLC Linked Data

### PR DESCRIPTION
**Pull request by courte**
This is a strategy for turning ISBNs into works by getting their title & author from OCLC. This PR moves to broader author extraction from OCLC Linked Data, which creates more data for the OCLC editions so that one of them can be used as a presentation edition.

**This is basically the opposite of our strategy with OCLC thus far** -- we're getting whatever data we can scrape together, including getting into evaluating author name quality, instead of trying to keep all of that information in its own little corner.

I think it needs:
- a better method for evaluating which edition from OCLC is the "best" or most accurate
- more nuanced author name evaluation

I also have a concern that if works created using this method are passed back into Circulation Managers, this data (with its Metadata Wrangler DataSource) could possibly be given priority over the data that Bibliotheca provides. I'm not sure where editions from the DataSource get placed in the spectrum.

I'm in the process of running this against 200 Bibliotheca-equivalent ISBNs to see what percentage get a work through this resolution process. On a much smaller sample size, I got 45% coverage.

A start to NYPL-Simplified/metadata_wrangler#66.

### CODE REVIEW

File: coverage.py [hunk 1]
```diff
@@ -196,13 +199,46 @@ def finalize(self, identifier):
 
         self.resolve_equivalent_oclc_identifiers(identifier)
         if identifier.type==Identifier.ISBN:
+            self.generate_edition(identifier)
             # Currently we don't try to create Works for ISBNs,
             # we just make sure all the Resources associated with the
             # ISBN are properly handled. At this point, that has
             # completed successfully, so do nothing.
```
Comment (leonardr) [inline coverage.py]: Is this comment still accurate?

File: tests/test_coverage.py [hunk 2]
```diff
@@ -161,6 +161,18 @@ def test_process_item_creates_license_pool(self):
...
+        # Prepare an identifier that already has a LicensePool through
+        # another source.
+        licensed = self._identifier(identifier_type=Identifier.OVERDRIVE_ID)
+        other_source = DataSource.lookup(self._db, DataSource.OVERDRIVE)
+        lp = LicensePool.for_foreign_id(
+            self._db, other_source, licensed.type, licensed.identifier
+        )[0]
+
+        self.coverage_provider.process_item(licensed)
+        eq_(lp, licensed.licensed_through)
```
Comment (leonardr) [inline tests/test_coverage.py]: I don't understand what this is testing. Are you verifying that `licensed_through` didn't get overwritten by some other value?

File: tests/test_oclc.py [hunk 3]
```diff
@@ -245,37 +268,43 @@ def test_all_authors_get_viaf_lookup(self):
...
+        # and not those who do not.
         lookup1 = (ContributorData(
                   viaf="1", display_name="Display Name",
                   family_name="Family", sort_name="Name, Sort",
                   wikipedia_name="Wikipedia_Name"), None, None)
         lookup2 = (ContributorData(
                    viaf="2", wikipedia_name="Robert_Jordan_(Author)",
                    biography="That guy."), None, None)
-        viaf.queue_lookup(lookup1, lookup2)
+        viaf.queue_lookup(lookup1, lookup2, "Unrequested lookup")
```
Comment (leonardr) [inline tests/test_oclc.py]: After you finish this test do you need to verify that 'unrequested lookup' is still stuck in the mock VIAF queue?

**Comment by courte** [comment]
When I ran 200 ISBNs, 124 of them ended up with a Work. Quality of title and author varies. It's not terrible, but getting the information directly from the source the way we discussed would be 💯.

File: tests/test_oclc.py (hunk 3 above)
Comment (courte) [inline tests/test_oclc.py]: I verify a few lines below (unless you're saying it's not necessary to do so?). In the latest commit I moved that verification and added a bit more of a comment to clarify.

Comment (leonardr) [inline tests/test_oclc.py]: ...don't know what I was thinking, you do verify it immediately afterwards. Never mind!

**Comment by courte** [comment]
@leonardr would you take another look at this next week? It still offers ~52% coverage (I did two more test runs, and I got 86/200 and 104/200 for two random batches of Bibliotheca-sourced ISBN equivalents).

When a Work isn't created, it's because the title and author couldn't be sourced from OCLC. Covers, descriptions, and summaries are still scraped together and returned as an entry.

**Pull request merged**